### PR TITLE
enh(dashboard): refactor the dataset section to support resource tree

### PR DESCRIPTION
**Pull request by Med1233**
## Description

 In the widget editon modal, when selecting my dataset (step 2), when adding a new filter the incompatible resources must not be proposed

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 22.04.x
- [ ] 22.10.x
- [ ] 23.04.x
- [ ] 23.10.x
- [ ] 24.04.x (master)

#### Community contributors & Centreon team

- [ ] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: centreon/www/front_src/src/Dashboards/SingleInstancePage/Dashboard/AddEditWidget/WidgetProperties/Inputs/Resources/useResources.ts [hunk 1]
```diff
@@ -212,8 +258,60 @@ const useResources = ({
     setFieldTouched(`data.${propertyName}`, true, false);
   };
 
+  const getCustomQueryParameters = (
+    index: number,
+    resourceType
+  ): Array<QueryParameter> => {
+    if (index < 1) {
```
Comment (Thebarda) [inline centreon/www/front_src/src/Dashboards/SingleInstancePage/Dashboard/AddEditWidget/WidgetProperties/Inputs/Resources/useResources.ts:265]:
```suggestion
    if (equals(index, 0)) {
```

File: centreon/www/front_src/src/Dashboards/SingleInstancePage/Dashboard/AddEditWidget/WidgetProperties/Inputs/Resources/useResources.ts [hunk 2]
```diff
@@ -169,6 +208,13 @@ const useResources = ({
 
   const changeResourceType =
     (index: number) => (e: ChangeEvent<HTMLInputElement>) => {
+      const selectedResourcesTypesCount = value?.length || 0;
+
+      if (selectedResourcesTypesCount - 1 > index) {
```
Comment (Thebarda) [inline centreon/www/front_src/src/Dashboards/SingleInstancePage/Dashboard/AddEditWidget/WidgetProperties/Inputs/Resources/useResources.ts:213]: Maybe extract it into a variable such as `isNotLastResourceTypeChanged`

File: centreon/www/front_src/src/Dashboards/SingleInstancePage/Dashboard/AddEditWidget/WidgetProperties/Inputs/Resources/Resources.tsx [hunk 3]
```diff
@@ -82,8 +87,8 @@ const Resources = ({
         <ItemComposition
           displayItemsAsLinked
           IconAdd={<AddIcon />}
-          addButtonHidden={!canEditField || singleResourceType}
-          addbuttonDisabled={!areResourcesFullfilled(value)}
+          addButtonHidden={iAddButtonHidden}
```
Comment (Thebarda) [inline centreon/www/front_src/src/Dashboards/SingleInstancePage/Dashboard/AddEditWidget/WidgetProperties/Inputs/Resources/Resources.tsx:90]:
```suggestion
          addButtonHidden={isAddButtonHidden}
```

File: centreon/www/front_src/src/Dashboards/SingleInstancePage/Dashboard/AddEditWidget/WidgetProperties/Inputs/Resources/Resources.tsx [hunk 4]
```diff
@@ -67,6 +68,10 @@ const Resources = ({
     (value.length <= 1 && (required || isNil(required))) ||
     equals(value.length, 1);
 
+  const iAddButtonHidden = !canEditField || singleResourceType;
```
Comment (Thebarda) [inline centreon/www/front_src/src/Dashboards/SingleInstancePage/Dashboard/AddEditWidget/WidgetProperties/Inputs/Resources/Resources.tsx:71]:
```suggestion
  const isAddButtonHidden = !canEditField || singleResourceType;
```

**Review by Thebarda — APPROVED** [review]

**Review by dmyios — APPROVED** [review]

**Pull request merged**